src/tlog/verify.ts

```diff
@@ -85,89 +87,152 @@ export function verifyTLogIntegratedTime(bundle: Bundle): void {
   });
 }
 
-// Returns a properly formatted "VerificationPayload" for one of the
-// transaction log entires in the given bundle which can be used for SET
-// verification.
-function toVerificationPayload(
-  bundle: Bundle,
-  publicKey: string,
-  index = 0
-): VerificationPayload {
-  // Ensure bundle has tlog entries
-  const entries = bundle.verificationData?.tlogEntries;
-  if (!entries || entries.length - 1 < index) {
-    throw new Error('No tlog entries found in bundle');
+export function verifyTLogBodies(bundle: Bundle): void {
+  bundle.verificationData?.tlogEntries.forEach((entry) => {
+    verifyTLogBody(entry, bundle);
+  });
+}
+
+// Compare the given intoto tlog entry to the given bundle
+function verifyTLogBody(entry: TransparencyLogEntry, bundle: Bundle): void {
+  if (!entry.kindVersion) {
+    throw new Error('no kindVersion found in bundle');
   }
 
-  // Rekor metadata
-  const { integratedTime, logIndex, logId } = entries[index];
+  const { kind, version } = entry.kindVersion;
+  const body: EntryKind = JSON.parse(entry.canonicalizedBody.toString('utf8'));
 
-  if (!logId) {
-    throw new Error('No logId found in bundle');
+  if (kind !== body.kind || version !== body.apiVersion) {
+    throw new Error('bundle content and tlog entry do not match');
   }
 
-  // Recreate the Rekor entry from the bundle
-  const body = toVerificationBody(bundle, publicKey);
+  switch (body.kind) {
+    case 'intoto':
+      verifyIntotoTLogBody(body, bundle);
+      break;
+    case 'hashedrekord':
+      verifyHashedRekordTLogBody(body, bundle);
+      break;
+    default:
+      throw new Error('Unknown kind found in tlog entry');
+  }
+}
 
-  return {
-    body: enc.base64Encode(json.canonicalize(body)),
-    integratedTime: Number(integratedTime),
-    logIndex: Number(logIndex),
-    logID: logId.keyId.toString('hex'),
-  };
+// Compare the given intoto tlog entry to the given bundle
+function verifyIntotoTLogBody(tlogEntry: IntotoKind, bundle: Bundle): void {
+  if (bundle.content?.$case !== 'dsseEnvelope') {
+    throw new Error('bundle content and tlog entry do not match');
```

Comment:
Should this error message say something else?
```suggestion
    throw new Error(`Unsupported bundle content: ${bundle.content?.$case || 'unknown'}`);
```